edb/server/multitenant.py

```diff
@@ -327,25 +336,39 @@ async def _add_tenant():
             async with self._tenants_lock[sni]:
                 if serial > self._tenants_serial.get(sni, 0):
                     self._tenants_conf.pop(sni, None)
+            metrics.mt_tenant_add_errors.inc(1.0, conf["instance-name"])
 
     async def _remove_tenant(self, serial: int, sni: str):
+        tenant = None
         try:
             async with self._tenants_lock[sni]:
                 if serial > self._tenants_serial.get(sni, 0):
                     if sni in self._tenants:
                         tenant = self._tenants.pop(sni)
+                        metrics.mt_tenant_remove_total.inc(
+                            1.0, tenant.get_instance_name()
+                        )
                         current_tenant.set(tenant.get_instance_name())
                         await self._destroy_tenant(tenant)
+                        metrics.mt_tenants_total.dec()
                         logger.info("Removed Tenant %s", sni)
                     self._tenants_serial[sni] = serial
         except Exception:
             logger.critical("Failed to remove Tenant %s", sni, exc_info=True)
+            if tenant is not None:
+                metrics.mt_tenant_remove_errors.inc(
+                    1.0, tenant.get_instance_name(),
+                )
```

Comment:
maybe increment the counter with `unknown` or something as the tenant name, if it's None? otherwise we have a spot where an exception can happen and fall through the cracks of our monitoring